### PR TITLE
Import Pixi as a peer dependency

### PR DESCRIPTION
**Pull request by getkey**
Otherwise it will possibly result in 2 different versions of Pixi getting installed by package managers, which breaks Pixi. More context [here](https://pixijs.slack.com/archives/CKTAAJM41/p1564856738006300).

### CODE REVIEW

File: tools/demo/src/filters/color-overlay.js [hunk 1]
```diff
@@ -1,9 +1,9 @@
...
-			folder.addColor(this, 'color');
-		}
-	});
+    this.addFilter('ColorOverlayFilter', {
+        fishOnly: true,
+        args: [0xff0000],
+        oncreate(folder) {
+            folder.addColor(this, 'color');
+        }
+    });
```
Comment (getkey) [inline tools/demo/src/filters/color-overlay.js]: `npm run build` did this automatically. I can remove this change from the PR if you want.

File: rollup.config.js [hunk 2]
```diff
@@ -139,9 +139,10 @@ async function main() {
         const input = path.join(basePath, 'src/index.js');
         const freeze = false;
 
-        // Generate the externals to use, by default don't include dependencies
-        const external = Object.keys(pkg.dependencies);
+        // Generate the externals to use, by default don't include peer dependencies
+        const external = Object.keys(pkg.peerDependencies || {});
```
Comment (bigtimebuddy) [inline rollup.config.js]: Not sure if this is correct. There are some packages (advanced-bloom) that still has a dependency and we should maintain those along with the peer dependencies.
```suggestion
        const external = [].concat(
            Object.keys(pkg.dependencies || {}),
            Object.keys(pkg.peerDependencies || {})
        );
```

File: rollup.config.js [hunk 3]
```diff
@@ -139,9 +139,10 @@ async function main() {
...
 
+        console.log(external,basePath,main,path);
```
Comment (bigtimebuddy) [inline rollup.config.js]: Remove
```suggestion
```

File: tools/demo/src/filters/color-overlay.js (hunk 1 above)
Comment (bigtimebuddy) [reply tools/demo/src/filters/color-overlay.js]: That's fine :+1:

**Comment by getkey** [comment]
Ping @bigtimebuddy. I experienced the issue again when updating from 5.2.0 to 5.2.1 :sweat_smile:

**Comment by bigtimebuddy** [comment]
Okay, I've been pondering about this for awhile, but I think you're right about the peerDependencies for core. I'll merge this soon.

**Pull request merged**